### DISCUSSION TITLE
multi-architecture?

### DISCUSSION THREAD

**Discussion by yaobiao131** [NetSparkleUpdater/NetSparkle#547] [General]
How does appcask.xml generate multi-platform and multi-architecture? The platform can use sparkle:os=windows. How to distinguish between multiple architectures? Should I add a sparkle:arch=x86,x64,arm64

**Comment by Deadpikle** [comment]
> How does appcast.xml generate multi-platform and multi-architecture?

It does not detect this from the binary. If there is no `OperatingSystem` (`-o`/`--os`) specified in the app cast generator, none will be put in. If, for example, you're on macOS and specify `--os Linux` for the app cast generator, it will assume that the binary directory you're using has binaries for Linux.

When the app itself is checking for updates, it will use `RuntimeInformation.IsOSPlatform` to see if the items in the app cast apply to the currently running software. If the app is running on `Windows` and the app cast specifies `sparkle:os=macos`, then that specific update will not be considered as a possible update to the file.

I'd recommend using separate app cast files per OS. It's just easier that way. If you want to do it in one giant app cast, you probably can with the right parameters to the app cast generator or by hand-building it.

> How to distinguish between multiple architectures?

`NetSparkleUpdater` currently does not consider the architecture of the binaries at any point. This would be a feature request if someone wanted to build that in. :-) If you want to do this for your application in the current version of NetSparkle, use separate app casts for each arch and detect which app cast URL to look at in your own app.
